lintreview/tools/eslint.py

```diff
@@ -36,12 +39,21 @@ def process_files(self, files):
         """Run code checks with ESLint.
         """
         log.debug('Processing %s files with %s', files, self.name)
+
+        if self.options.get('install'):
+            output = run_command(
+                ['npm', 'install'],
```

Comment:
My concern with installing application dependencies is that many react/ember/angular applications have build dependencies that are in the 100-400MB range. Downloading all those dependencies on each review will add a significant runtime cost and potentially create bandwidth usage problems.

Reply:
It's optional, so the user can decide as they see fit.

Reply:
Also, with node 8 and package-lock.json, installs are much quicker than they were in the past.

Reply:
Also, installing the project's dependencies was how I resolved the issue of the linter generating erroneous "unresolved reference to PACKAGE" errors. I suppose one could disable that linting rule, but then you'd miss real cases of unresolved references.

One way to deal with the extra bandwidth might be to put in more logic around cleanup. Instead of deleting the repo after every lint, maybe it's deleted when the PR is closed or merged? Maybe there's a configurable cleanup frequency (e.g. every day, week, etc.)?

Reply:
I think the npm cache actually takes care of the issue of bandwidth, to a large degree